integration-test/src/test/java/org/sagebionetworks/ITStorageLocation.java

```diff
@@ -196,7 +221,150 @@ public void synapseS3StorageLocationWithSts() throws SynapseException {
 
 		testStsStorageLocation(synapseS3StorageLocationSetting);
 	}
-	
+
+	@Test
+	public void sampleCode_MigrateS3Bucket() throws Exception {
+		// Create external S3 storage location with STS.
+		ExternalS3StorageLocationSetting externalS3StorageLocationSetting = createExternalS3StorageLocation();
+		externalS3StorageLocationSetting.setStsEnabled(true);
+		externalS3StorageLocationSetting = synapse.createStorageLocationSetting(externalS3StorageLocationSetting);
+		applyStorageLocationToFolder(externalS3StorageLocationSetting);
+
+		// Create some test data to migrate
+		// baseKey/f1
+		// baseKey/f2
+		// baseKey/a/f3
+		// baseKey/a/f4
+		// baseKey/a/b/c/f5
+		// baseKey/a/b/c/f6
+		String baseKey = externalS3StorageLocationSetting.getBaseKey();
+		uploadToExternalS3Bucket(baseKey, "f1", "sample content 1");
+		uploadToExternalS3Bucket(baseKey, "f2", "sample content 2");
+		uploadToExternalS3Bucket(baseKey + "/a", "f3", "sample content 3");
+		uploadToExternalS3Bucket(baseKey + "/a", "f4", "sample content 4");
+		uploadToExternalS3Bucket(baseKey + "/a/b/c", "f5", "sample content 5");
+		uploadToExternalS3Bucket(baseKey + "/a/b/c", "f6", "sample content 6");
+
+		// Run sample code.
+		MigrateS3Bucket migration = new MigrateS3Bucket(synapseS3Client.getUSStandardAmazonClient(), synapse,
+				externalS3Bucket, baseKey, folder.getId(), externalS3StorageLocationSetting.getStorageLocationId());
+		migration.execute();
+
+		// Verify results.
+		// Get files in baseKey (f1, f2, and owner.txt).
+		Map<String, EntityHeader> childrenByName = getChildren(folder.getId(), EntityType.file);
+		assertEquals(3, childrenByName.size());
+		assertTrue(childrenByName.containsKey("f1"));
+		assertTrue(childrenByName.containsKey("f2"));
+		assertTrue(childrenByName.containsKey("owner.txt"));
```

Comment:
Make a comment in the sample code that we migrate owner.txt. You can add a check to prevent this.